_posts/2017-10-04-scalamacros.md

```diff
@@ -0,0 +1,299 @@
+- in Scala 2.13, we continue to experiment via compiler plugins and compiler
+  feature flags in later minor releases
+- in Scala 2.14 macros v3 become no longer “experimental” and scala.reflect is
+  deprecated
+
+Following the recommendation of the Scala Center Advisory Board, the work on
+macros v3 will be an iterative processes between
+
+-  implementing macro features that have been approved for inclusion into
+   macros v3
+-  gathering feedback from the community on what macro features merit inclusion
+   in macros v3
+
+As for the first part, we immediately begin development to support a limited
+subset of blackbox def macros.
+
+### Blackbox def macros
+
+Blackbox def macros have non-controversially proven themselves to be invaluable
+for the Scala community.
+Examples of blackbox def macros include: `CaseClass.fieldNames` above,
+ScalaTest `assert` and Play/Circe JSON automatic readers/writers.
+Blackbox def macros share the following attributes:
+
+- they're invoked at compile-time instead of runtime
+- they can query the compiler for semantic information such as types and symbols.
+- they faithfully respect their declared type signatures, making their
+  implementation irrelevant to understand their behavior. From the end user's
+  perspective, they look and behave much like regular Scala methods excluding
+  the ability to emit compiler errors/warnings
+
+These attributes of blackbox def macros enable them to mix naturally into
+Scala codebases and make them prime for inclusion in the Scala Language
+Specification.
+
+### SIP proposal
+
+Alongside prototyping preliminary support for a limited set of blackbox def
+macros, we will immediately begin preparing a SIP proposal to include macros
+v3 into the Scala Language Specification.
+We plan to address the valuable reviews made to [SIP-29] on inline/meta in a
+new proposal, so that SIP-29 can be rejected.
+In addition, we will document how we we aim to solve hygiene using an
+innovation discovered by the collaboration of Liu Fengyun and Eugene Burmako.
+Hygiene plagues most macro tutorials and was previously considered to be out of
+scope for SIP-29
+
+### Documentation
+
+Currently, there is no official guide for exploring the wide landscape of
+metaprogramming facilities that are available to Scala developers.
+While discussing macros with members of the OSS community and industry,
+I repeatedly receive questions such as:
+should I use def macros, annotations, shapeless, a compiler plugin or scripted
+code generation?
+There is no silver bullet, each solution comes with a set of trade-offs.
+
+We plan to document the trade-offs of these different metaprogramming
+solutions.
+Especially, we want to document which use-cases macros are suitable for, and
+which use-cases they're either overkill, or insufficient.
+Macros should in general be used as a last resort.
+We care deeply to know what you'd like to see covered in this documentation
+effort.
+Please share your thoughts.
```

Comment:
Are you going to link to a Scala Contributors thread here?

Reply:
Yes, as soon as they are up I will open a PR adding links.